geograypher/meshes/meshes.py

```diff
@@ -162,19 +164,57 @@ def load_mesh(
             simplify_tol_meters=ROI_simplify_tol_meters,
         )
 
-        # Downsample mesh if needed
+        # Downsample mesh if needed transfer textures from original mesh to downsampled mesh
         if downsample_target != 1.0:
             # TODO try decimate_pro and compare quality and runtime
             # TODO see if there's a way to preserve the mesh colors
             # TODO also see this decimation algorithm: https://pyvista.github.io/fast-simplification/
             self.logger.info("Downsampling the mesh")
-            self.pyvista_mesh = self.pyvista_mesh.decimate(
+            intermediate_mesh = self.pyvista_mesh.decimate(
                 target_reduction=(1 - downsample_target)
             )
+            self.pyvista_mesh = self.transfer_texture(intermediate_mesh)
         self.logger.info("Extracting faces from mesh")
         # See here for format: https://github.com/pyvista/pyvista-support/issues/96
         self.faces = self.pyvista_mesh.faces.reshape((-1, 4))[:, 1:4].copy()
 
+    def transfer_texture(self, downsampled_mesh):
+        """Transfer texture from original mesh to a downsampled version using KDTree for nearest neighbor point searches
+
+        Args:
+            downsampled_mesh (pv.PolyData): The downsampled version of the original mesh
+
+        Returns:
+            pv.PolyData: The downsampled mesh with the transferred textures
+        """
+        # Only compute KDTree if active scalars are associated with points
+        if (
+            self.kdtree is None
+            and self.pyvista_mesh.active_scalars_info.association
+            == pv.FieldAssociation.POINT
+        ):
+            # Store original mesh points in KDTree for nearest neighbor search
+            self.kdtree = KDTree(self.pyvista_mesh.points)
+
+            # For ecah point in the downsampled mesh find the nearest neighbor point in the original mesh
+            _, indices = self.kdtree.query(downsampled_mesh.points)
+
+            # Get active scalars name and value at the found indices
+            active_scalars_name = self.pyvista_mesh.active_scalars_name
+            active_scalars = self.pyvista_mesh.active_scalars[indices]
```

Comment:
This line could be a little confusing, maybe add some more details here and/or change the variable names to be slightly more descriptive. For example, including the term downsample/subsampled would make sense to me.

Reply:
Also, it's completely reasonable to only do the active scalars for now, but if you wanted, you could leave a TODO to consider transferring all the point based scalars.